etcdserver/api/v3rpc/maintenance.go

```diff
@@ -128,11 +128,11 @@ func (ms *maintenanceServer) Snapshot(sr *pb.SnapshotRequest, srv pb.Maintenance
 }
 
 func (ms *maintenanceServer) Hash(ctx context.Context, r *pb.HashRequest) (*pb.HashResponse, error) {
```

Comment:
The expectation for Hash(rev=0) now is that etcd is in a quiescent state (and retry a few times in case the server hasn't settled), so it shouldn't need to worry about the compaction revision. There are events aside from compaction that can cause hash mismatches for rev=0 (e.g., lease expiration, member reconfiguration), so the extra logic to wait on the compaction doesn't buy anything in terms of hash stability.

HashRev(rev=0) should hash the keys bucket using the current rev (if it's worth supporting, there's no way to access this functionality if there's only going to be one Hash RPC), not hash all the buckets; Hash() already does that.